react/features/base/responsive-ui/middleware.native.js

```diff
@@ -19,65 +13,19 @@ let handler;
  * @param {Store} store - The redux store.
  * @returns {Function}
  */
-MiddlewareRegistry.register(store => next => action => {
+MiddlewareRegistry.register(({ dispatch }) => next => action => {
     const result = next(action);
 
     switch (action.type) {
-    case APP_WILL_UNMOUNT: {
-        _appWillUnmount();
+    case CLIENT_RESIZED: {
+        const { clientWidth: width, clientHeight: height } = action;
+
+        dispatch(setAspectRatio(width, height));
+        dispatch(setReducedUI(width, height));
```

Comment:
Why do we need to intercept the CLIENT_RESIZED action type and dispatch the exact same parameters twice? Wouldn't it be easier to read/understand to intercept CLIENT_RESIZED right at the place the values are used?

Reply:
I thought of it, but went for leaving less code changed. Will fix!

Reply:
Actually, it's not going to be that simple. The reduced UI action is intercepted in another middleware to request a lower layer. So I'm inclined to leave it as is, since I don't have a way to get rid of both.